python/src/Collection_doc.i.in

```diff
@@ -67,7 +67,14 @@ Examples
 [Normal(mu = 0, sigma = 1),Uniform(a = -0.5, b = 1)]
 >>> vector.add(ot.Gamma(2.75, 1.0))
 >>> print(vector)
-[Normal(mu = 0, sigma = 1),Uniform(a = -0.5, b = 1),Gamma(k = 2.75, lambda = 1, gamma = 0)]"
+[Normal(mu = 0, sigma = 1),Uniform(a = -0.5, b = 1),Gamma(k = 2.75, lambda = 1, gamma = 0)]
+
+
+- Collection of **design of experiments**:
```

Comment:
This is true, and this is on purpose. @jschueller argue that the Python user can use square brackets (or round braces) to create a `Collection` so that showing this type of object is not useful. I added that page to the index, and then removed it following @jschueller's advice. So if you agree, I would stick to the present content.

Reply:
Still, the current content is inconsistent in this regard. Shall I remove the whole python/src/Collection_doc.i.in file?

Reply:
Let us consider the following code snippet:

```python
dis = ot.ComposedDistribution([ot.Uniform()]*3)
col = dis.getDistributionCollection()
type(col)      
```
Output:
```python
openturns.model_copula.DistributionCollection
```
IMO we need some documentation for `Collection` types or else users could be confused by these object types. I see 2 possibilities:

1. either document all `Collection` types (`DistributionCollection`, `BoolCollection`, `MetropolisHastingsCollection`, etc.) or
2. have only one documentation page for the generic `Collection` concept.

The second solution seems better to me. Currently it seems to me we are in a "worst of all worlds" situation where some `XxxCollection` classes are documented, but not all, and the generic `Collection` concept is not.

What I would therefore do is index `Collection` in [python/doc/user_manual/base_objects.rst](https://github.com/openturns/openturns/blob/master/python/doc/user_manual/base_objects.rst#vectors) and remove `XxxCollection` API entries (or maybe move their documentation to the generic `Collection` page).

Reply:
Actually, the current situation is more complicated than this. Here is a copy of the current API entry (not in the PR, but from the official doc) :
![image](https://user-images.githubusercontent.com/31351465/160017758-27a029a7-eb1b-471f-943f-3649a7ddbbf2.png)
What you see is that a couple of `xCollection` are documented, but not all. The documented classes are: BoolCollection, IndicesCollection, ComplexCollection, ScalarCollection. Other types of collections exist, but are undocumented. This is the case for the `DistributionCollection` you mention. 

The generic `Collection` class does not, _however_, have a doc. This is because there is no such generic class: all `xCollection` classes are produced from a template. In the particular case of `WeightedCollection`, the code is:
https://github.com/openturns/openturns/blob/94c4806a9030408908fdd7795a32a0097fdf1db6/lib/src/Uncertainty/Algorithm/WeightedExperiments/openturns/TensorProductExperiment.hxx#L42

The situation gets a little confused actually.

Reply:
I don't think it matters that there is no actual "Collection" class. There is a `python/src/Collection.i` file, and a `python/src/Collection_doc.i.in` file, so I believe (but I may be mistaken) a "Collection" HTML page could be produced. It is currently not produced because "Collection" is not indexed in any `python/doc/user_manual` RST toctree file, hence my suggestion in the first comment of this conversation:


> I don't think `python/src/Collection_doc.i.in` produces any HTML doc page: this may be because it is not indexed in [python/doc/user_manual/base_objects.rst](https://github.com/openturns/openturns/blob/master/python/doc/user_manual/base_objects.rst#vectors).

IMO that would be more efficient than having to decide whether we want to create a documentation page for any or all `XxxCollection` objects.

Reply:
I work on this and will commit an update ASAP.

Reply:
Adding a `Collection` item to [python/doc/user_manual/base_objects.rst](https://github.com/openturns/openturns/blob/master/python/doc/user_manual/base_objects.rst#vectors) does not seem to add a `Collection` entry in the doc. I guess that this is because there is no `Collection` class, so that the SWIG trickery in [python/src/Collection.i](https://github.com/openturns/openturns/blob/4153ef7cd7b0e9cabc87602721d5e5324d7ad274/python/src/Collection.i) does not work as intended.

Reply:
Actually, there is no shared point of view on the method to document the `Collection` classes. This is why I remove this at commit e319d49a7c4187bdb961598dc5f0c335cfd138e5

Reply:
This is the issue #1999. 
Is this OK for you @josephmure ?

Reply:
No problem. This is only tangentially related to this PR anyway. Could you perhaps keep this conversation "unresolved" so we can check it later, when this PR is merged?

Reply:
Ok. I let this open.